src/test/run-make/issue-40535/Makefile

```diff
@@ -3,8 +3,8 @@
 # * `bar` declares and depends on `extern crate baz`
 # * All crates built in metadata-only mode (`cargo check`)
 all:
-	$(RUSTC) baz.rs --emit=metadata
-	$(RUSTC) bar.rs --emit=metadata --extern baz=libbaz.rmeta
-	$(RUSTC) foo.rs --emit=metadata -L dependency=. --extern bar=libbar.rmeta 2>&1 | \
+	$(RUSTC) baz.rs --emit=metadata --out-dir=$(TMPDIR)
```

Comment:
It may need to be after `--emit=metadata`, because that's how Cargo writes the command.